app/main/lib/text_similarity.py

```diff
@@ -201,3 +208,8 @@ def search_text_by_model(search_params):
     return {
         'result': response
     }
+
+def retrieve_openai_embeddings(text, model_key):
+    openai.api_key = app.config['OPENAI_API_KEY']
```

Comment:
i'm more familiar with `app.config.OPENAI_API_KEY` style syntax for referencing public class variables. (because then the linter will tell me if I get the key wrong! :-)

Reply:
If this is making an external API call at this point, maybe worth logging here so we have a timestamp to refer to in debugging (maybe openai utils allready do that)